evm_arithmetization/src/generation/prover_input.rs

```diff
@@ -92,23 +92,26 @@ impl<F: Field> GenerationState<F> {
     fn run_trie_ptr(&mut self, input_fn: &ProverInputFn) -> Result<U256, ProgramError> {
         let trie = input_fn.0[1].as_str();
         match trie {
-            "state" => match self.trie_root_ptrs.state_root_ptr {
-                Some(state_root_ptr) => Ok(state_root_ptr),
-                None => {
-                    let mut new_content = self.memory.get_preinit_memory(Segment::TrieData);
-
-                    let n = load_state_mpt(&self.inputs.trimmed_tries, &mut new_content)?;
-
-                    self.memory.insert_preinitialized_segment(
-                        Segment::TrieData,
-                        crate::witness::memory::MemorySegmentState {
-                            content: new_content,
-                        },
-                    );
-                    Ok(n)
-                }
-            }
-            .map(U256::from),
+            "initial_state" => self
+                .trie_root_ptrs
+                .state_root_ptr
+                .map_or_else(
```

Comment:
Is there a possibility that we reach the default case? In the kernel this function is only called when checking the initial state at the end of the execution, so there should always be a state root pointer.

Reply:
Yes, and it corresponds to the `None` case before. The trie is not written in memory until you call this function. Actually it's the other branch that's never called because you only access the inital trie once, IIRC.